SpockOpaleye/src/UserApi.hs

```diff
@@ -17,49 +17,46 @@ module UserApi
   , deactivateUser
   ) where
 
-import           ApiBase
+import           AppCore
 import           Control.Arrow
 import           Control.Lens
 import           Control.Monad.IO.Class
 import           Data.Maybe
-import           DataTypes
 import           GHC.Int
 import           Opaleye
-import           OpaleyeDef
 
-import           CryptoDef
 import           Prelude                    hiding (id)
 
-createUser :: UserIncoming -> AppM (Auditable User)
+createUser :: UserIncoming -> AppM User
 createUser user = do
   Just hash <- liftIO $ bcryptPassword $ user ^. password
   let fullUser = user { _userpolyPassword = hash }
-  auditable <$> (createRow userTable fullUser)
+  createRow userTable fullUser
 
-updateUser :: Auditable User -> AppM (Auditable User)
+updateUser :: User -> AppM User
 updateUser user = updateAuditableRow userTable user
 
-activateUser :: Auditable User -> AppM (Auditable User)
+activateUser :: User -> AppM User
 activateUser user = setUserStatus user UserStatusActive
 
-deactivateUser :: Auditable User -> AppM (Auditable User)
+deactivateUser :: User -> AppM User
 deactivateUser user = setUserStatus user UserStatusInActive
 
-setUserStatus :: Auditable User -> UserStatus -> AppM (Auditable User)
+setUserStatus :: User -> UserStatus -> AppM User
 setUserStatus user newStatus = updateUser $ user & status .~ newStatus
```

Comment:
shall we follow a convention to put an underscore at the ned of such internal APIs: `setUserStatus_`? So that if they are exported by mistake, it's easier to spot.